src/screens/VaultScreen.js

```diff
@@ -109,14 +150,40 @@ export default class VaultScreen extends React.Component {
   /**
    * Upon call, returns the filter category elements based on the category array in the state.
    */
-  upvote(itemId) {
-    var resourcesArr = this.state.resources.slice()
-    for(var i = 0; i < resourcesArr.length; i++) {
-      if(resourcesArr[i].id == itemId) {
-        resourcesArr[i].upvotes = resourcesArr[i].upvotes+1;
+  async upvote(resourceId, hasUpvoted) {
+    const veteranId = this.props.navigation.state.params.id;
+    if (hasUpvoted) {
+      try {
+        const endpoint = APIRoutes.deleteUpvote();
+    		const params = {
+    			upvote: {
+    				resource_id: resourceId,
+    				veteran_id: veteranId
+    			}
+    		};
+        let response_json = await BaseRequester.patch(endpoint, params);
+        this.retrieveResources();
```

Comment:
does this reload all the resources for a single upvote change?

Reply:
yes, it resets the resources displayed

Reply:
do we need to reset all the resources if the veteran made 1 upvote?

Reply:
@lbkchen @amillman is there a way to reflect the changes to a single upvoted resource without re-rendering all the resources?

Reply:
is it ok to just reflect it by triggering a style change?

Reply:
resolved